dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/validation/ValidationRunner.java

```diff
@@ -25,40 +25,36 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.hisp.dhis.dxf2.events.importer.delete.preprocess;
-
-import static org.hisp.dhis.importexport.ImportStrategy.DELETE;
+package org.hisp.dhis.dxf2.metadata.objectbundle.validation;
 
 import java.util.List;
 import java.util.Map;
-import java.util.function.Predicate;
 
-import lombok.Getter;
-import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 
-import org.hisp.dhis.dxf2.events.importer.AbstractProcessorFactory;
-import org.hisp.dhis.dxf2.events.importer.ImportStrategyUtils;
-import org.hisp.dhis.dxf2.events.importer.Processor;
+import org.hisp.dhis.common.IdentifiableObject;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundle;
+import org.hisp.dhis.feedback.TypeReport;
 import org.hisp.dhis.importexport.ImportStrategy;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
-/**
- * @author Luciano Fiandesio
- */
-@Getter
-@Component( "eventsPreDeleteProcessorFactory" )
+@Component
 @RequiredArgsConstructor
-public class PreDeleteProcessorFactory extends AbstractProcessorFactory
+class ValidationRunner
 {
 
-    @NonNull
-    @Qualifier( "eventDeletePreProcessorMap" )
-    private final Map<ImportStrategy, List<Class<? extends Processor>>> processorMap;
+    private final Map<ImportStrategy, List<ValidationCheck>> validatorMap;
 
-    private final ImportStrategy importStrategy = DELETE;
+    public <T extends IdentifiableObject> TypeReport executeValidationChain( ObjectBundle bundle, Class<T> klass,
```

Comment:
Just noticed the `TypeReport` return type here. These are not "cheap" to create and in a success case we just want to return a placeholder saying that there is nothing to report. I switched this around for metadata validation, see `org.hisp.dhis.dxf2.metadata.objectbundle.validation.ObjectValidationCheck` so that `check` returns `void` and potential things to report are collected by having a `Consumer<ObjectReport> addReports` parameter. Maybe you could do something similar here?

Reply:
Maybe we can improve this part if necessary in a separate PR